src/Stryker.CLI/Stryker.CLI/CommandLineConfigHandler.cs

```diff
@@ -0,0 +1,195 @@
+                    CommandOptionType.MultipleValue => HandleMultiValue(cliInput, (IInputDefinition<IEnumerable<string>>)strykerInput),
+                    CommandOptionType.SingleOrNoValue => HandleSingleOrNoValue(strykerInput, cliInput, inputs),
+                    _ => true
+                };
+
+                _ = strykerInput switch
+                {
+                    IInputDefinition inputDefinition when inputDefinition.GetType().BaseType.GetGenericArguments()[0] == typeof(string) => HandleSingleStringValue(cliInput, (IInputDefinition<string>)inputDefinition),
+                    IInputDefinition inputDefinition when inputDefinition.GetType().BaseType.GetGenericArguments()[0] == typeof(int?) => HandleSingleIntValue(cliInput, (IInputDefinition<int?>)inputDefinition),
+                    IInputDefinition inputDefinition when inputDefinition.GetType().BaseType.GetGenericArguments()[0] == typeof(int) => HandleSingleIntValue(cliInput, (IInputDefinition<int?>)inputDefinition),
+                    _ => true
+                };
+            }
+        }
+
+        private bool HandleNoValue(IInputDefinition<bool?> strykerInput)
+        {
+            strykerInput.SuppliedInput = true;
+            return true;
+        }
+
+        private bool HandleSingleStringValue(CommandOption cliInput, IInputDefinition<string> strykerInput)
+        {
+            strykerInput.SuppliedInput = cliInput.Value();
+            return true;
+        }
+
+        private bool HandleSingleIntValue(CommandOption cliInput, IInputDefinition<int?> strykerInput)
+        {
+            if (int.TryParse(cliInput.Value(), out int value))
+            {
+                strykerInput.SuppliedInput = value;
+            } else
+            {
+                throw new StrykerInputException($"Unexpected value for argument {cliInput.LongName}:{cliInput.Value()}. Expected type to be integer");
+            }
+            return true;
+        }
+
+        private bool HandleSingleOrNoValue(IInputDefinition strykerInput, CommandOption cliInput, IStrykerInputs inputs)
+        {
+            switch (strykerInput)
+            {
+                // handle single or no value inputs
+                case SinceInput sinceInput:
+                    sinceInput.SuppliedInput = true;
+                    inputs.SinceTargetInput.SuppliedInput = cliInput.Value();
+                    break;
+                case WithBaselineInput withBaselineInput:
+                    withBaselineInput.SuppliedInput = true;
+                    inputs.BaselineProviderInput.SuppliedInput = cliInput.Value();
+                    break;
+            }
+            return true;
+        }
+
+        private bool HandleMultiValue(CommandOption cliInput, IInputDefinition<IEnumerable<string>> strykerInput)
+        {
+            strykerInput.SuppliedInput = cliInput.Values;
+
+            return true;
```

Comment:
Why do these methods need to even return anything? The values seem to be discarded anyway.